soc/riscv/riscv-privilege/andes_v5/pma.c

```diff
@@ -237,7 +249,10 @@ static int pma_init(const struct device *arg)
 
 		__ASSERT(0, "CPU doesn't support PMA. "
 			    "Please disable CONFIG_SOC_ANDES_V5_PMA\n");
-		return -1;
+#ifndef CONFIG_ASSERT
+		LOG_ERR("CPU doesn't support PMA. Please disable CONFIG_SOC_ANDES_V5_PMA");
```

Comment:
Andes PMA is use to configure nocache memory region, but DMA sample depends on CONFIG_NOCACHE_MEMORY rather than PMA status, so we think this is error to run incompatible Kconfig with CPU feature.

Reply:
In that case - shouldn't this error out in the compile time instead of the runtime? Because this is something that we will know won't work without having to run it on hw/in sim. Or is there any advantage to doing this check in the runtime?

Reply:
Ok, fair enough. Thanks for your explanation.